tests/lib/fortigate_test.py

```diff
@@ -0,0 +1,352 @@
+
+    no_interfaces_acl = fortigate.Fortigate(
+        policy.ParsePolicy(GOOD_HEADER + no_interfaces_term,
+                           self.naming), EXP_INFO)
+    src_only_int_acl = fortigate.Fortigate(
+        policy.ParsePolicy(GOOD_HEADER + src_only_int_term,
+                           self.naming), EXP_INFO)
+    dest_only_int_acl = fortigate.Fortigate(
+        policy.ParsePolicy(GOOD_HEADER + dest_only_int_term,
+                           self.naming), EXP_INFO)
+    both_interfaces_acl = fortigate.Fortigate(
+        policy.ParsePolicy(GOOD_HEADER + both_interfaces_term,
+                           self.naming), EXP_INFO)
+
+    no_interfaces_sig = 'set srcintf any\n\t\tset dstintf any'
+    src_int_only_sig = 'set srcintf wan1\n\t\tset dstintf any'
+    dest_int_only_sig = 'set srcintf any\n\t\tset dstintf wan2'
+    both_interfaces_sig = 'set srcintf wan1\n\t\tset dstintf wan2'
+
+    self.assertIn(
+        no_interfaces_sig, str(no_interfaces_acl),
+        '[%s]' % str(no_interfaces_acl))
+    self.assertIn(
+        src_int_only_sig, str(src_only_int_acl),
+        '[%s]' % str(src_only_int_acl))
+    self.assertIn(
+        dest_int_only_sig, str(dest_only_int_acl),
+        '[%s]' % str(dest_only_int_acl))
+    self.assertIn(
+        both_interfaces_sig, str(both_interfaces_acl),
+        '[%s]' % str(both_interfaces_acl))
+
+  def testLogging(self):
+    log_term = self.fmt.format(TERM_TEMPLATE,
+                               logging='true')
+    no_log_term = self.fmt.format(TERM_TEMPLATE,
+                                  remove_fields=('logging',))
+
+    log_acl = fortigate.Fortigate(
+        policy.ParsePolicy(GOOD_HEADER + log_term,
+                           self.naming), EXP_INFO)
+    no_log_acl = fortigate.Fortigate(
+        policy.ParsePolicy(GOOD_HEADER + no_log_term,
+                           self.naming), EXP_INFO)
+
+    log_sig = 'set logtraffic all'
+
+    self.assertIn(
+        log_sig, str(log_acl), '[%s]' % str(log_acl))
+    self.assertNotIn(
+        log_sig, str(no_log_term), '[%s]' % str(no_log_acl))
+
+  def testDuplicateTermError(self):
+    term = self.fmt.format(TERM_TEMPLATE, logging='true')
+    duplicate_terms = term + term
+    parsed_p = policy.ParsePolicy(GOOD_HEADER + duplicate_terms,
+                                  self.naming)
+
+    self.assertRaises(fortigate.FortiGateDuplicateTermError,
+                      fortigate.Fortigate,
+                      parsed_p,
+                      EXP_INFO)
+
+  def testPortMap(self):
+    port_map = fortigate.FortigatePortMap()
+    self.assertEqual('SSH', port_map.get_protocol('tcp', 22))
+    self.assertRaises(fortigate.FortiGatePortDoesNotExistError,
+                      port_map.get_protocol,
+                      'tcp', 5000)
+    self.assertRaises(fortigate.FortiGateValueError,
+                      port_map.get_protocol,
+                      'bad_proto', 22)
```

Comment:
These tests are never actually run it seems. You forgot to include the if main section with a call to the unittests.